test/utest.cpp

```diff
@@ -80,17 +81,34 @@ private slots:
       QVERIFY(active_states.size() > 0);
 
       // Get the available events for the first active state
-      const QSet<QString> available_events = map.at(active_states.at(0));
+      const std::set<std::pair<QString, QString>> available_events = map.at(active_states.at(0));
       if (available_events.empty())
       {
         std::cout << "State '" << active_states.at(0).toStdString() << "' has no available events" << std::endl;
         break;
       }
 
+      // Get the available events for the first active state
+      std::set<std::pair<QString, QString>> action_ids = sm.getStateTransitionMap().at(active_states.first());
+      if (action_ids.empty())
+      {
+        std::cout << "State '" << active_states.at(0).toStdString() << "' has no available events" << std::endl;
+        break;
+      }
+
+      QList<QString> events;
+      QList<QString>::iterator k;
+      for (auto pair : action_ids)
+      {
+        events.insert(k, pair.first);
+        k += 1;
+      }
+
       // Choose a random event to submit
       static std::mt19937 gen(1);
       std::uniform_int_distribution<int> dist(0, available_events.size() - 1);
-      const QString event = available_events.toList().at(dist(gen));
+
+      const QString event = events.at(dist(gen));
```

Comment:
If you are wanting to select a random index in a `std::set` it would probably be easier to do something like:
```c++
const QString event = std::next(available_events.begin(), dist(gen)).first;
```

Reply:
so there's a change on 93 and then on 111 according to this suggested edit - am I just replacing the line on 93 with the content from your comment ?

Reply:
@marip8 still unsure on this one